sbpy/units/core.py

```diff
@@ -125,3 +128,148 @@ def spectral_density_vega(wfb):
+    M_sun : `~astropy.units.Quantity`
+        Solar magnitude in the same magnitude system as the quantity to be
+        converted.  If `wfb` is not provided, then `M_sun` will be used
+        in the conversion.  If neither `wfb` nor `M_sun` is present, then
+        the default V-band solar magnitude, -26.775 VEGAmag will be used.
+
+    Returns
+    -------
+    eqv : list
+        List of equivalencies
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from astropy import units as u
+    >>> from sbpy.units import magnitude_reflectance, VEGAmag
+    >>> m = 3.4 * VEGAmag
+    >>> xsec = np.pi * (460 * u.km)**2
+    >>> ref = m.to('1/sr', magnitude_reflectance(xsec))
+    >>> print('{0:.4f}'.format(ref))
+    0.0287 1 / sr
+    >>>
+    >>> m1 = ref.to(VEGAmag, magnitude_reflectance(xsec))
+    >>> print('{0:.2f}'.format(m1))
+    3.40 mag(VEGA)
+    """
+    if wfb is not None:
+        try:
+            import synphot
+        except ImportError:
+            warn(AstropyWarning('synphot required for Vega-based magnitude'
+                            ' conversions.'))
+            return []
+        sun = Sun.from_default()
+        if isinstance(wfb, u.Quantity):
+            f_sun = sun(wfb).to(VEGA, spectral_density_vega(wfb))
+        elif isinstance(wfb, (synphot.SpectralElement, str)):
+            f_sun = sun.filt(wfb)[1].to(VEGA, spectral_density_vega(wfb))
+        else:
+            raise ValueError('unrecognized type for `wfb`')
+        M_sun = f_sun.to(VEGAmag)
+        f_sun = f_sun.value
+    else:
+        if M_sun is None:
+            M_sun = -26.77471503 * VEGAmag
+        if not hasattr(M_sun.unit, 'physical_unit'):
+            raise u.UnitTypeError('the magnitude system must be based on a physical unit')
+        f_sun = M_sun.to(M_sun.unit.physical_unit).value
+    return [(M_sun.unit.physical_unit,
+             u.sr**-1,
+             lambda mag: mag/(f_sun*xsec.to('au2').value),
+             lambda ref: ref*f_sun*xsec.to('au2').value)]
+
+
+def magnitude_xsection(ref, wfb=None, M_sun=None):
+    """Magnitude - cross-section equivalencies to convert between magnitude
+    and scattering cross-section for given average reflectance.
+
+    The default magnitude system is ``VEGAmag``, where the apparent magnitude
+    of Vega is assumed to be 0 in all and any wavelengths and bands.  If other
+    magnitude system is used, then it is implicitly inferred from the solar
+    magnitude passed by keyword 'M_sun'.
+
+    Parameters
+    ----------
+    ref : `~astropy.units.Quantity`
+        Average reflectance of the object of interest
+    wfb : `~astropy.units.Quantity`, `~synphot.SpectralElement`, string
+        Wavelength, frequency, or a bandpass of the corresponding flux
+        density being converted.  See
+        :func:`~synphot.SpectralElement.from_filter()` for possible
+        bandpass names.  If provided, then this parameter overrides `M_sun`.
```

Comment:
What should be the behavior for multiple wavelengths?  I think it should use the `Sun.observe` method in #124, which would treat a set of wavelengths as the bin centers of a spectrum, rather than interpolate the solar spectrum to each wavelength.  Whichever the case, the assumption should be documented in the docstring.

Reply:
Agreed.  Will wait for #124 to be merged before finalize this PR.

Reply:
@mkelley I admit I'm confused here.  Why would we need to worry about the case of multiple wavelength?  The quantity to be converted is always observed at one single wavelength or band, right?  Furthermore, when multiple wavelengths/bands are passed, `Sun.observe` would return multiple flux density values.  How would we use multiple values, average them?

Reply:
I was thinking about the case of a low-resolution spectrum.  Each element in the spectrum is the average flux density over a small wavelength range.  The solar spectrum should also be averaged over the small wavelength range.  Interpolation to the central wavelength could be wrong.  This matters most near absorption features a higher resolution solar spectra.

Reply:
With a single wavelength, all we can ever do is interpolate the solar spectrum and the user must accept the possibility of a bias.  In such a case a bandpass would be more appropriate.

Back to the spectrum example.  If you want to convert a Deep Impact spectrum of the surface of Hartley 2 to a reflectance spectrum, I think it should be allowed to pass all wavelengths at once, rather than iterating through each element.

Reply:
OK I got it.  Do you have any good spectral data for testing?

Reply:
How about Hartley 2?

Reply:
Tempel 1 will be easier.